### PR TITLE
Use Dask's `sort_values` for first column sorting in `apply_sort`

### PR DESCRIPTION
**Pull request by charlesbluca**
Now that Dask's `sort_values` has support for ascending/descending and null positioning, we can use it to do what `_sort_first_column` was once doing in a (hopefully) more performant way. There are still some failures here that are resolved with https://github.com/dask/dask/pull/8225, so this should be safe to merge by the next Dask release.

### CODE REVIEW

**Comment by rjzamora** [comment]
>cc @rjzamora if you have any thoughts on this

I agree that much of the dask_cudf sort_values logic should live in upstream dask.dataframe. However, I will need to think a bit more about the `sort_function` idea to provide any useful feedback.  

**Somewhat-related thoughts**: One thing I like about the dask_cudf logic is that `sort_values` uses `rearrange_by_column` rather than `rearrange_by_divisions`.  This makes the mutli-column sorting code path a bit simpler, because we can use multiple columns to define the new (temporary) "_partitions" column, but then the rest of the logic is the same as a single-column sort. I also like that dask_cudf's set_index is directly based on sort_values (which is not the case in dask.dataframe).

**Comment by charlesbluca** [comment]
Looks like the original fix I had in mind for dask/dask#8255 is actually needed here if we intend to use upstream Dask sorting functions (`rearrange_by_divisions`, `set_partitions_pre`) directly on a dask-cudf dataframe. However, I'm more interested in seeing if we can move most of the logic for sorting upstream, maybe by allowing for a custom sorting function in dask/dask-cudf's `sort_values`. This would allow for the same performance benefits of using the internal sorting logic instead of a `sort_values` call, while also keeping things pretty simple downstream.

In any case, I'm pretty sure using `sort_values` instead of `_sort_first_column` for the initial sort is still beneficial performance-wise, so we can merge this now and make a follow up depending on what direction we decide to move in.

File: dask_sql/physical/utils/sort.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def apply_sort(
             by=sort_columns,
             ascending=sort_ascending,
             na_position="first" if sort_null_first[0] else "last",
-        )
+        ).persist()
```
Comment (charlesbluca) [inline dask_sql/physical/utils/sort.py:28]: Noting that I added this persist call after sorting to match up with the other sort paths, which persist after sorting.

File: dask_sql/physical/utils/sort.py [hunk 2]
```diff
@@ -37,34 +37,30 @@ def apply_sort(
         and not any(sort_null_first)
     ):
         try:
-            return df.sort_values(sort_columns, ignore_index=True)
+            return df.sort_values(sort_columns, ignore_index=True).persist()
```
Comment (charlesbluca) [inline dask_sql/physical/utils/sort.py:40]: Same as above

**Pull request merged**